### PR TITLE
Updated device SDK documentation

### PR DESCRIPTION
**Pull request by celanthe**
Updated device SDK documentation with information on  two new SDK APIs we introduced in this sprint: APN config & disable power off via Samsung Knox.

### CODE REVIEW

File: docs/home/devicesdk.md [hunk 1]
```diff
@@ -37,7 +36,6 @@ Before any operation can be done with the SDK, it needs to be initialized first.
 EsperDeviceSDK sdk = EsperDeviceSDK.getInstance(getApplicationContext());
```
Comment (celanthe) [inline docs/home/devicesdk.md:36]: @ankitgupta29 @mihirEsper Hi! What is the updated information needed in the implementation files dependencies section to bring lines 27-28 up to date with the Tessarion_MR8 2.0.6 release?

Thank you so much.

Comment (mihirEsper) [reply docs/home/devicesdk.md:36]: replace `esperdevicesdk-release-v1.2.4` with `app-2.0.6225.9`

File: docs/home/devicesdk.md [hunk 2]
```diff
@@ -356,3 +358,159 @@ sdk.enableWifiTethering(​"EsperSDKHotspot"​, ​"123123123"​, true, ​new
...
+
+* Once power off is disabled, powering off is enabled only if another call is made to re-enable it. 
+* If powering off is disabled, a toast notification with the message "Security policy prevents power off" appears when the user tries to power off the device.
+
+**This API is only available on Samsung KNOX enabled devices & is available from the Esper Device SDK version TESSARION_MR8.**
+
+The function call to allowPowerOff returns boolean if allowing or disallowing power off was successful.
+
+```java
+sdk.allowPowerOff(checked, new EsperDeviceSDK.Callback<Boolean>() {
```
Comment (mihirEsper) [inline docs/home/devicesdk.md:373]: replace checked with `true/false`.

File: docs/home/devicesdk.md [hunk 3]
```diff
@@ -356,3 +358,159 @@ sdk.enableWifiTethering(​"EsperSDKHotspot"​, ​"123123123"​, true, ​new
...
+       showFailureResult(t);
+   }
+});
+```
+
+### Configure APNs via the Esper Device SDK
+
+APIs to add/update/remove/setDefault APN confuguraton(s).
+
+These APIs are only available on Samsung KNOX enabled devices, via Supervisor plugin, and the Esper Device SDK version TESSARION_MR8.
```
Comment (mihirEsper) [inline docs/home/devicesdk.md:391]: Samsung KNOX enabled devices OR via Supervisor plugin

File: docs/home/devicesdk.md [hunk 4]
```diff
@@ -356,3 +358,159 @@ sdk.enableWifiTethering(​"EsperSDKHotspot"​, ​"123123123"​, true, ​new
...
+
+To create new / update an existing APN, a JSON string needs to be passed in SDK functions with APN config parameters mentioned.
+
+```java
```
Comment (mihirEsper) [inline docs/home/devicesdk.md:395]: shouldn't it be JSON?

File: docs/home/devicesdk.md [hunk 5]
```diff
@@ -356,3 +358,159 @@ sdk.enableWifiTethering(​"EsperSDKHotspot"​, ​"123123123"​, true, ​new
...
+
+           @Override
+           public void onFailure(Throwable t) {
+               Log.e(TAG, "onFailure: ", t);
+               showFailureResult(t);
+           }
+}, apnID);
+```
+
+removeApnConfig expects APN ID in argument returned by addNewApnConfig function.
```
Comment (mihirEsper) [inline docs/home/devicesdk.md:513]: repalce `removeApnConfig ` with `setDefaultApn`

**Review by mihirEsper — CHANGES_REQUESTED** [review]

File: docs/home/devicesdk.md (hunk 5 above)
Comment (celanthe) [reply docs/home/devicesdk.md:513]: Done, will be pushing the commit with this change shortly.

File: docs/home/devicesdk.md (hunk 4 above)
Comment (celanthe) [reply docs/home/devicesdk.md:395]: I've changed all instances to be JSON instead of Java, thanks for the tip!

File: docs/home/devicesdk.md (hunk 3 above)
Comment (celanthe) [reply docs/home/devicesdk.md:391]: Made this change, thank you!

File: docs/home/devicesdk.md (hunk 2 above)
Comment (celanthe) [reply docs/home/devicesdk.md:373]: Change made!

File: docs/home/devicesdk.md (hunk 1 above)
Comment (celanthe) [reply docs/home/devicesdk.md:36]: Thank you so much!

**Review by mihirEsper — CHANGES_REQUESTED** [review]
JSON had to be used only from line# 396-419. rest of of the places `java` correct & does not require a change

**Comment by celanthe** [comment]
> JSON had to be used only from line# 396-419. rest of of the places `java` correct & does not require a change

Hi @mihirEsper thanks for letting me know the exact lines to have that change apply to. I've updated the rest of the instances back to Java!

**Review by mihirEsper — APPROVED** [review]
lgtm!

**Pull request merged**